assets/src/application/ui/forms/espressoForm/components/index.ts

```diff
@@ -0,0 +1,3 @@
+export { default as DatePicker } from './DatePicker';
+
+export { default as TimePicker } from './TimePicker';
```

Comment:
how come these are not part of  `application/ui/forms/espressoForm/adapters/index.ts ` ?

Reply:
Adapters are for RFF `<->` AntD integration. These are simple Date/Time picker components that don't use `momentjs` like those of AntD pickers. They use vanila JS `Date`. These components can be used anywhere either in `EspressoForm` or even in EDTR.

Reply:
this might seem kinda convoluted, but we should use adapters for both RFF and AntD, ie:

- AntD adapters to decouple us from that lib
- RFF adapters that wrap the above components with the Field component

the basic AntD adapters could go in `assets/src/application/ui/input/`
and the RFF Field adapters could go in `assets/src/application/ui/forms/espressoForm/adapters/`

reason being is that it is possible we will need inputs from time to time that are not going to be integrated with RFF

Reply:
It's not possible to create an adapter for RFF alone because RFF suplies `FieldRenderProps` which you have to pass down to the actual input component, in our case AntD components.

Reply:
```js
const AntdAdapter = ({ name, value, etc, ...rest }) => {
    return <AntdComponent name={name} value={value} etc={etc} {...rest} />;
}

const rffAdapter = ({ ...rest }) => {
    return <Field component={ AntdAdapter } {...rest} />;
}
```

how would something similar to the above not work?


just looked again at what you have going on in:

- `\espressoForm\adapters` and 
- `\espressoForm\adapters\MappedField.tsx`
- `\espressoForm\fields\Field.tsx`
- `\espressoForm\renderers\FieldRenderer.tsx`

and it's basically the same thing as my example above... basically using spread operators to pass thru any additional props.
You're doing everything dynamically though, which i think is great, but the only thing that would need to change would be the import path in `MappedField` for all of the base components. Those could simply come from `assets/src/application/ui/input/`

Reply:
I will try that approach to see how it works. Thanks 👍

Reply:
agree, we might need to pass raw fields at some point in some specific cases or edge cases, so it would be nice to have a little bit of decoupling between rff and antd